File: Solutions/2596/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var _ = Console.ReadLine(); //ignore num of inputs
            var divisors = GetInputs<int>(' ');

            var counter = 0;
            for (var number = 1; number <= 1000; number++) {
                if (divisors.All(divisor => number % divisor == 0))
                    counter++;
            }

            Console.WriteLine(counter);
        }

        private static List<T> GetInputs<T>(char separator) =>
            Console.ReadLine()
                ?.Trim()
                .Split(separator)
                .Select(item => (T) Convert.ChangeType(item, typeof(T))).ToList();
    }
}

File: Solutions/2596/python3/program.py
from typing import List


def can_valid(number: int, divisors: List[int]) -> bool:
    for divisor in divisors:
        if number % divisor != 0:
            return False
    return True


def main():
    _ = input()
    divisors = list(map(int, input().split()))

    count = 0
    for i in range(1, 1001):
        if can_valid(i, divisors):
            count += 1

    print(count)


if __name__ == '__main__':
    main()
